File: simulation_prototype/Receipt.py
class Receipt:
    def __init__(self, new_orderID, new_productList):
        self._ID = new_orderID
        self._orderID = new_orderID
        self._isPaid = False
        self._productList = new_productList
        self._totalPrice = 0

        self._countPrice()


    def getID(self):
        return self._ID


    def getOrderID(self):
        return self._orderID


    def isPaid(self):
        return self._isPaid


    def paidReceipt(self):
        self._isPaid = True


    def getTotalPrice(self):
        return self._totalPrice


    def _countPrice(self):
        for product in self._productList:
            self._totalPrice += product.getPrice()

File: Receipt.cpp

#include "Receipt.h"

Receipt::Receipt(int new_orderID, std::vector<Product*> new_productList)
{
    ID = new_orderID;
    orderID = new_orderID;
    isPaid = false;    
    totalPrice = 0;
    
    setProducts(new_productList);
}

int Receipt::getID()
{
    return ID;
}

int Receipt::getOrderID()
{
    return orderID;
}

bool Receipt::getIsPaid()
{
    return isPaid;
}

void Receipt::paidReceipt()
{
    isPaid = true;
}

int Receipt::getTotalPrice()
{
    return totalPrice;
}

void Receipt::setProducts(std::vector<Product*> new_productList)
{
    for (auto product_ptr : new_productList)
    {
        productList.push_back(product_ptr);

        totalPrice += product_ptr -> getPrice();
    }
}

Receipt::~Receipt()
{

}